src/models/ontology_mapping_engine.py

```diff
@@ -0,0 +1,128 @@
+from src.models import ontology_mapper_st as oms
+from src.models import ontology_mapper_lm as oml
+from OmicsMLRepoHarmonizer.src.models import ontology_mapper_rag as ollm
+import src.CustomLogger.custom_logger
+import pandas as pd
+import numpy as np
+from thefuzz import fuzz
+
+logger = src.CustomLogger.custom_logger.CustomLogger()
+
+
+
+class OntoMapEngine():
+    def __init__(self, 
+                 method:str, 
+                 query:list[str], 
+                 corpus:list[str],
+                 cura_map:dict, 
+                 topk:int=5, 
+                 yaml_path:str='method_model.yaml', 
+                 om_strategy='st') -> None:
+        """
+        This class is responsible for initializing the OntoMapEngine object.
+
+        Args:
+            method (str): The name of the method.
+            query (list): The list of queries.
+            corpus (list): The list of corpus.
+            topk (int): The number of top matches to return.
+            from_tokenizer (bool): A flag to indicate if the model is loaded from a tokenizer.
+            yaml_path (str): The path to the YAML file.
+        """
+        self.method = method
+        self.query = query
+        self.corpus = corpus
+        self.topk = topk
+        self.om_strategy = om_strategy
+        # self.from_tokenizer = from_tokenizer
+        self.yaml_path = yaml_path
+        self.cura_map = cura_map
+
+        self._logger = logger.custlogger(loglevel='INFO')
+        self._logger.info("Initialized OntoMap Engine module")
+
+
+    def _exact_matching(self):
+        """
+        This function is responsible for performing exact matching.
+
+        Here we want to separate exact matches of query to large corpus 
+        Returns:
+            list: The list of exact matches from query 
+        """
+        return [q for q in self.query if q in self.corpus] 
```

Comment:
Make `_exact_matching` not-case-sensitive (fuzz doesn't seem to cover this well)

Reply:
I will add this in the later versions.